plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSourceProvider.java

```diff
@@ -92,7 +93,8 @@ private List<ColumnAdaptation> getColumnAdaptations(Optional<List<JdbcColumnHand
         if (mergeRowIdChannel == scanColumns.size()) {
             columnAdaptationBuilder.add(buildMergeIdColumnAdaptation(scanColumns, mergeRowIdColumn));
         }
-        return columnAdaptationBuilder.build();
+        // Trim the pages. Jdbc connectors need to follow the required columns strictly.
+        return columnAdaptationBuilder.build().subList(0, columnCount);
```

Comment:
why did this work for phoenix?

Reply:
One difference is in the `buildSql`, the base jdbc needs to determine the columns names,types to build the sql to get source, but the phoenix use it own api to get result set.